goal_src/levels/swamp/swamp-rat-nest.gc

```diff
@@ -5,3 +5,4199 @@
+                                         0
+                                         frame-group
+                                         )
+                                        )
+     num-func-seek!
+     )
+    )
+   (until (ja-done? 0)
+    (suspend)
+    (let ((a0-1 (-> self skel root-channel 0)))
+     (set!
+      (-> a0-1 param 0)
+      (the float (+ (-> a0-1 frame-group data 0 length) -1))
+      )
+     (set! (-> a0-1 param 1) 1.0)
+     (joint-control-channel-group-eval!
+      a0-1
+      (the-as art-joint-anim #f)
+      num-func-seek!
+      )
+     )
+    )
+   (go swamp-rat-nest-dummy-idle)
+   (none)
+   )
+  :post
+  (the-as (function none :behavior swamp-rat-nest-dummy) transform-post)
+  )
+
+(defstate swamp-rat-nest-dummy-die (swamp-rat-nest-dummy)
+  :event
+  (the-as
+   (function process int symbol event-message-block object :behavior swamp-rat-nest-dummy)
+   #f
+   )
+  :code
+  (behavior ()
+   (set!
+    (-> self state-time)
+    (the-as seconds (-> *display* base-frame-counter))
+    )
+   (set!
+    (-> self parent-process 0 dummy)
+    (the-as (pointer swamp-rat-nest-dummy) #f)
+    )
+   (let ((gp-0 (get-process *default-dead-pool* part-tracker #x4000)))
+    (when gp-0
+     (let ((t9-1 (method-of-type part-tracker activate)))
+      (t9-1
+       (the-as part-tracker gp-0)
+       *entity-pool*
+       'part-tracker
+       (the-as pointer #x70004000)
+       )
+      )
+     (run-now-in-process
+      gp-0
+      part-tracker-init
+      (-> self death-part)
+      -1
+      #f
+      #f
+      #f
+      (->
+       self
+       node-list
+       data
+       (-> self particle-spawn-joint)
+       bone
+       transform
+       vector
+       3
+       )
+      )
+     (-> gp-0 ppointer)
+     )
+    )
+   (until
+    (>=
+     (- (-> *display* base-frame-counter) (the-as int (-> self state-time)))
+     30
+     )
+    (suspend)
+    )
+   (ja-channel-set! 0)
+   (while (-> self child)
+    (suspend)
+    )
+   (deactivate self)
+   (none)
+   )
+  )
+
+(defbehavior
+  swamp-rat-nest-dummy-init-by-other swamp-rat-nest-dummy
+  ((arg0 swamp-rat-nest-dummy))
+  (let ((gp-0 (-> arg0 entity)))
+   (set! (-> self mask) (logior (process-mask enemy) (-> self mask)))
+   (dummy-20 self)
+   (process-drawable-from-entity! self gp-0)
+   (set! (-> self top-sphere quad) (-> self root-override trans quad))
+   (set! (-> self top-sphere y) (+ 24576.0 (-> self top-sphere y)))
+   (set! (-> self top-sphere w) 18432.0)
+   (set! (-> self entity) gp-0)
+   )
+  (set!
+   (-> self nav)
+   (new 'process 'nav-control (-> self root-override) 16 40960.0)
+   )
+  (logior!
+   (-> self nav flags)
+   (nav-control-flags display-marks bit3 bit5 bit6 bit7)
+   )
+  (let* ((gp-1 (-> self nav))
+         (s5-0 (method-of-object gp-1 dummy-14))
+         )
+   (let ((t9-3 (method-of-object (-> self nav) dummy-18)))
+    (-> self root-override trans)
+    (t9-3)
+    )
+   (s5-0 gp-1)
```

Comment:
I think this is a nested method call and `dummy-14` probably takes one more argument